keel-orca/src/main/kotlin/com/netflix/spinnaker/keel/constraints/PipelineConstraintEvaluator.kt

```diff
@@ -67,6 +69,23 @@ class PipelineConstraintEvaluator(
 
     val judge = "pipeline:${constraint.pipelineId}"
 
+    val currentVersion = keelRepository
+      .getCurrentVersionInEnv(deliveryConfig, targetEnvironment.name, artifact.reference)
+
+    if (currentVersion != version) {
+      repository
+        .storeConstraintState(
+          state.copy(
+            status = FAIL,
+            comment = "Pipeline constraint wants to run against $version, " +
+              "but the current version in env ${targetEnvironment.name} is $currentVersion.",
+            judgedAt = clock.instant(),
+            judgedBy = "keel"
+          )
+        )
+      return false
+    }
```

Comment:
@lorin should we fail the constraint here? I thought it was an ok thing to do, since this pipeline constraint will get replaced soon. But I'm not sure.